closure/goog/array/array.js

```diff
@@ -346,20 +346,33 @@ goog.array.map = goog.NATIVE_ARRAY_PROTOTYPES &&
 goog.array.reduce = goog.NATIVE_ARRAY_PROTOTYPES &&
                     (goog.array.ASSUME_NATIVE_FUNCTIONS ||
                      goog.array.ARRAY_PROTOTYPE_.reduce) ?
-    function(arr, f, val, opt_obj) {
-      goog.asserts.assert(arr.length != null);
-      if (opt_obj) {
-        f = goog.bind(f, opt_obj);
+  function(arr, f, val, opt_obj) {
+    goog.asserts.assert(arr.length != null);
+    var params = [];
+    for(var i = 1, l = arguments.length; i < l; i++){
```

Comment:
Gotcha, I understand.  FWIW, the current implementation fails most basic use-cases of reduce, e.g.,

```
// sum an array of numbers
goog.array.reduce([1,2], function(a,b){
  return a+b;
});
```

^ Fails, whereas the native implementation (and my patch) returns 3, as expected.

Not the end of the world, but definitely not a drop in replacement for Array.prototype.reduce.